extensions/sql-database-projects/src/dialogs/deployDatabaseQuickpick.ts

```diff
@@ -0,0 +1,142 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as constants from '../common/constants';
+import { AppSettingType, IDeployProfile, ILocalDbSetting } from '../models/deploy/deployProfile';
+import { Project } from '../models/project';
+import * as generator from 'generate-password';
+import { getPublishDatabaseSettings } from './publishDatabaseQuickpick';
+let path = require('path');
+let fse = require('fs-extra');
+
+/**
+ * Create flow for Deploying a database using only VS Code-native APIs such as QuickPick
+ */
+export async function launchDeployDatabaseQuickpick(project: Project): Promise<IDeployProfile | undefined> {
+
+	// Show options to user for deploy to existing server or docker
+
+	const deployOption = await vscode.window.showQuickPick(
+		[constants.deployToExistingServer, constants.deployToDockerContainer],
+		{ title: constants.selectDeployOption, ignoreFocusOut: true });
+
+	// Return when user hits escape
+	if (!deployOption) {
+		return undefined;
+	}
+
+	let localDbSetting: ILocalDbSetting | undefined;
+	// Deploy to docker selected
+	if (deployOption === constants.deployToDockerContainer) {
+		let portNumber = await vscode.window.showInputBox({
+			title: constants.enterPortNumber,
+			ignoreFocusOut: true,
+			value: constants.defaultPortNumber,
+			validateInput: input => isNaN(+input) ? constants.portMustNotBeNumber : undefined
+		}
+		);
+
+		// Return when user hits escape
+		if (!portNumber) {
+			return undefined;
+		}
+
+		let password: string | undefined = generator.generate({
+			length: 10,
+			numbers: true,
+			symbols: true,
+			lowercase: true,
+			uppercase: true,
+			exclude: '`"\'' // Exclude the chars that cannot be included in the password. Some chars can make the command fail in the terminal
+		});
+		password = await vscode.window.showInputBox({
```

Comment:
If they choose the generated password how are they supposed to use it later?

Reply:
password will be saved for the connection using vscode password store